src/index.tsx

```diff
@@ -28,17 +28,22 @@ function domElementGetter(): HTMLElement {
   // Make sure there is a div for us to render into
   let el = document.getElementById(pluginName);
   if (!el) {
-    el = document.createElement("div");
+    el = document.createElement('div');
   }
 
   return el;
 }
 
+// This was throwing a warning that rootComponent was not found even if it was defined
+// Defining loadRootComponent did not throw this warning
+// May be worth investigating further
 const reactLifecycles = singleSpaReact({
   React,
   ReactDOM,
-  rootComponent: App,
+  // rootComponent: App,
   domElementGetter,
+  loadRootComponent: () =>
+    new Promise((resolve, reject) => resolve(() => <App />)),
```

Comment:
I don't see why we need to use `loadRootComponent` here - and I think it would be good as well to move the error boundary stuff from `App` to here

Reply:
@louise-davies Made the changes